server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/Mailbox.scala

```diff
@@ -97,8 +114,15 @@ sealed trait QuotasExtension extends MailboxExtensionAdditionalFields {
   def quotas: Quotas
 }
 
-object Mailbox {
-  type MailboxName = String Refined NonEmpty
+object MailboxName {
+  type MailboxNameConstraint = NonEmpty
+  type MailboxName = String Refined MailboxNameConstraint
+
+  def liftOrThrow(value: String): MailboxName =
+    refined.refineV[MailboxNameConstraint](value) match {
+      case scala.util.Right(value) => value
```

Comment:
Note: I had to put the full path here, or the compiler was taking that as the `Right` final case class we wrote in `Rights` file